### PR TITLE
Machine ID: Bot Instance heartbeating

### PR DESCRIPTION
**Pull request by strideynet**
Related to https://github.com/gravitational/teleport/issues/41456

Introduces an implementation of the SubmitHeartbeat RPC and the code necessary within tbot to submit a heartbeat.

No changelog since we'll do one big changelog entry when backporting this entire feature.

```yaml
kind: bot_instance
metadata:
  expires:
    nanos: 905422000
    seconds: 1720098731
  name: 9abc2c84-7fb3-46bb-aa11-cce389b727ab
  revision: 0298c6a0-7655-4cd6-ab17-5887fb990674
spec:
  bot_name: testbot8383
  instance_id: 9abc2c84-7fb3-46bb-aa11-cce389b727ab
status:
  initial_authentication:
    authenticated_at:
      nanos: 916478000
      seconds: 1720084008
    generation: 1
    join_method: token
    join_token: '************************cfa52fba'
    public_key: LS0tLS1CRUdJTiBSU0EgUFVCTElDIEtFWS0tLS0tCk1JSUJJakFOQmdrcWhraUc5dzBCQVFFRkFBT0NBUThBTUlJQkNnS0NBUUVBMG42RXpNakxkNlYvb2M0MnZtUlEKd2RNMi9TYnl2VzJNV3B6Z1lvMEE0REVhKzVoM3pwcS9WcGtFWXNFQ2VqMmRlMTVyRU9sdHNPR3lXSXVjYlgvUwp5ZFR2TmRwTUdMZy8zOWEwYXlvaktndTUyb1BDS1ZpTlVITHVCaVBVdEoxZHFIcHFmRVFhemZnbTg1eXB3bU5jCk1VWEwzc3FUTFkrS2Z0c2xHcjN6QjBWOUdQU0tsOHNJckZ2cjF6SmdqbDg1QklKRFh3OWZvRm0ybXk2UGxxbGcKRWFoNzZ5YzBKbTlxRWlhMjRRV3hhb1ZLb001TXNkMUUyaXMxM0kzc2RYVnNZNUcvTXlaZVcwOWVBeU5PTEY2dwpVMzlZQlU1dmhrWUpaS1A4T0h4QUhxN1BlM3I1WUFBaWdrdFRMcEhqblhzMUxJdG45dndXVU1Yb3FPNTlZVUZXCmp3SURBUUFCCi0tLS0tRU5EIFJTQSBQVUJMSUMgS0VZLS0tLS0K
  initial_heartbeat:
    architecture: arm64
    hostname: Noahs-MacBook-Pro.local
    is_startup: true
    join_method: token
    one_shot: true
    os: darwin
    recorded_at:
      nanos: 4455000
      seconds: 1720084009
    uptime:
      nanos: 235452750
    version: 17.0.0-dev
  latest_authentications:
  - authenticated_at:
      nanos: 916478000
      seconds: 1720084008
    generation: 1
    join_method: token
    join_token: '************************cfa52fba'
    public_key: LS0tLS1CRUdJTiBSU0EgUFVCTElDIEtFWS0tLS0tCk1JSUJJakFOQmdrcWhraUc5dzBCQVFFRkFBT0NBUThBTUlJQkNnS0NBUUVBMG42RXpNakxkNlYvb2M0MnZtUlEKd2RNMi9TYnl2VzJNV3B6Z1lvMEE0REVhKzVoM3pwcS9WcGtFWXNFQ2VqMmRlMTVyRU9sdHNPR3lXSXVjYlgvUwp5ZFR2TmRwTUdMZy8zOWEwYXlvaktndTUyb1BDS1ZpTlVITHVCaVBVdEoxZHFIcHFmRVFhemZnbTg1eXB3bU5jCk1VWEwzc3FUTFkrS2Z0c2xHcjN6QjBWOUdQU0tsOHNJckZ2cjF6SmdqbDg1QklKRFh3OWZvRm0ybXk2UGxxbGcKRWFoNzZ5YzBKbTlxRWlhMjRRV3hhb1ZLb001TXNkMUUyaXMxM0kzc2RYVnNZNUcvTXlaZVcwOWVBeU5PTEY2dwpVMzlZQlU1dmhrWUpaS1A4T0h4QUhxN1BlM3I1WUFBaWdrdFRMcEhqblhzMUxJdG45dndXVU1Yb3FPNTlZVUZXCmp3SURBUUFCCi0tLS0tRU5EIFJTQSBQVUJMSUMgS0VZLS0tLS0K
  - authenticated_at:
      nanos: 905980000
      seconds: 1720084031
    generation: 2
    join_method: token
    public_key: c3NoLXJzYSBBQUFBQjNOemFDMXljMkVBQUFBREFRQUJBQUFCQVFEU2ZvVE15TXQzcFgraHpqYStaRkRCMHpiOUp2SzliWXhhbk9CaWpRRGdNUnI3bUhmT21yOVdtUVJpd1FKNlBaMTdYbXNRNlcydzRiSllpNXh0ZjlMSjFPODEya3dZdUQvZjFyUnJLaU1xQzduYWc4SXBXSTFRY3U0R0k5UzBuVjJvZW1wOFJCck4rQ2J6bktuQ1kxd3hSY3ZleXBNdGo0cCsyeVVhdmZNSFJYMFk5SXFYeXdpc1crdlhNbUNPWHprRWdrTmZEMStnV2JhYkxvK1dxV0FScUh2ckp6UW1iMm9TSnJiaEJiRnFoVXFnemt5eDNVVGFLelhjamV4MWRXeGprYjh6Smw1YlQxNERJMDRzWHJCVGYxZ0ZUbStHUmdsa28vdzRmRUFlcnM5N2V2bGdBQ0tDUzFNdWtlT2RlelVzaTJmMi9CWlF4ZWlvN24xaFFWYVAK
  latest_heartbeats:
  - architecture: arm64
    hostname: Noahs-MacBook-Pro.local
    is_startup: true
    join_method: token
    one_shot: true
    os: darwin
    recorded_at:
      nanos: 4455000
      seconds: 1720084009
    uptime:
      nanos: 235452750
    version: 17.0.0-dev
  - architecture: arm64
    hostname: Noahs-MacBook-Pro.local
    is_startup: true
    join_method: token
    one_shot: true
    os: darwin
    recorded_at:
      nanos: 957177000
      seconds: 1720084031
    uptime:
      nanos: 143367459
    version: 17.0.0-dev
version: v1
```

### CODE REVIEW

File: lib/tlsca/ca.go [hunk 1]
```diff
@@ -184,6 +184,9 @@ type Identity struct {
 	// BotName indicates the name of the Machine ID bot this identity was issued
 	// to, if any.
 	BotName string
+	// BotInstance indicates the name of the Machine ID Bot Instance this
+	// identity is associated with.
+	BotInstance string
```
Comment (strideynet) [inline lib/tlsca/ca.go:189]: TODO: This is temporary until another PR is opened that adds the proper field here.

**Comment by strideynet** [comment]
DNM until other PRs this relies on are merged and this is rebased.

**Comment by strideynet** [comment]
Ready for review - I've merged the bot instance heartbeating in tbot into this PR so it's a more cohesive chunk

**Review by timothyb89 — APPROVED** [review]
Working great on my test cluster!

**Review by espadolini — APPROVED** [review]

**Pull request merged**